### PR TITLE
[FLINK-24307][docs] Some lines in Scala code example end with unneces…

### PR DESCRIPTION
**Pull request by paul8263**
…sary semicolons. Also some Scala code blocks are wrongly typed with Java blocks

## What is the purpose of the change

Some lines in Scala code example end with unnecessary semicolons. Also some Scala code blocks are wrongly typed with Java blocks.

## Brief change log

- docs/content/docs/dev/dataset/transformations.md
- docs/content/docs/dev/datastream/event-time/generating_watermarks.md
- docs/content/docs/dev/datastream/operators/joining.md
- docs/content/docs/dev/datastream/operators/overview.md
- docs/content/docs/dev/datastream/testing.md
- docs/content/docs/dev/datastream/user_defined_functions.md
- docs/content/docs/dev/table/sql/alter.md
- docs/content/docs/dev/table/sql/create.md
- docs/content/docs/dev/table/common.md
- docs/content/docs/dev/table/data_stream_api.md
- docs/content/docs/dev/table/tableApi.md
- docs/content/docs/dev/table/types.md
- docs/content.zh/docs/dev/dataset/transformations.md
- docs/content.zh/docs/dev/datastream/event-time/generating_watermarks.md
- docs/content.zh/docs/dev/datastream/operators/joining.md
- docs/content.zh/docs/dev/datastream/operators/overview.md
- docs/content.zh/docs/dev/datastream/testing.md
- docs/content.zh/docs/dev/datastream/user_defined_functions.md
- docs/content.zh/docs/dev/table/sql/alter.md
- docs/content.zh/docs/dev/table/sql/create.md
- docs/content.zh/docs/dev/table/common.md
- docs/content.zh/docs/dev/table/data_stream_api.md
- docs/content.zh/docs/dev/table/tableApi.md
- docs/content.zh/docs/dev/table/types.md

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

File: docs/content.zh/docs/dev/table/data_stream_api.md [hunk 1]
```diff
@@ -788,8 +788,8 @@ val table =
                 Schema.newBuilder()
                     .column("uid", DataTypes.TINYINT())
                     .column("payload", DataTypes.STRING())
-                    .build())
-            .build());
+                  .build())
```
Comment (slinkydeveloper) [inline docs/content.zh/docs/dev/table/data_stream_api.md:791]: `.build()` should be aligned with the rest of the builder methods invocation

File: docs/content.zh/docs/dev/table/data_stream_api.md [hunk 2]
```diff
@@ -788,8 +788,8 @@ val table =
...
+          .build())
```
Comment (slinkydeveloper) [inline docs/content.zh/docs/dev/table/data_stream_api.md:792]: Same

File: docs/content/docs/dev/table/data_stream_api.md [hunk 3]
```diff
@@ -788,8 +788,8 @@ val table =
                 Schema.newBuilder()
                     .column("uid", DataTypes.TINYINT())
                     .column("payload", DataTypes.STRING())
-                    .build())
-            .build());
+                  .build())
+          .build())
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/table/data_stream_api.md:792]: Same as above

**Review by slinkydeveloper — COMMENTED** [review]
Left some minor comments, but overall looks good to me

**Comment by paul8263** [comment]
Hi @slinkydeveloper ,
Those enclosing brackets and indentation issues were fixed. Thanks for reviewing.

**Comment by paul8263** [comment]
@flinkbot run azure

File: docs/content/docs/dev/datastream/operators/joining.md [hunk 4]
```diff
@@ -277,11 +277,9 @@ orangeStream
     .intervalJoin(greenStream.keyBy(elem => /* select key */))
     .between(Time.milliseconds(-2), Time.milliseconds(1))
     .process(new ProcessJoinFunction[Integer, Integer, String] {
-        override def processElement(left: Integer, right: Integer, ctx: ProcessJoinFunction[Integer, Integer, String]#Context, out: Collector[String]): Unit = {
-         out.collect(left + "," + right); 
-        }
-      });
-    });
+        override def processElement(left: Integer, right: Integer, ctx: ProcessJoinFunction[Integer, Integer, String]#Context, out: Collector[String]): Unit = out.collect(left + "," + right)
```
Comment (zentol) [inline docs/content/docs/dev/datastream/operators/joining.md:280]: please move it back to a separate line

File: docs/content/docs/dev/dataset/transformations.md [hunk 5]
```diff
@@ -1195,8 +1195,8 @@ public class PointWeighter
 }
 
 DataSet<Tuple2<String, Double>>
-            weightedRatings =
-            ratings.join(weights) // [...]
+        weightedRatings =
+        ratings.join(weights); // [...]
```
Comment (zentol) [inline docs/content/docs/dev/dataset/transformations.md:1199]:
```suggestion
        ratings.join(weights);
```

File: docs/content.zh/docs/dev/dataset/transformations.md [hunk 6]
```diff
@@ -1197,8 +1197,8 @@ public class PointWeighter
 }
 
 DataSet<Tuple2<String, Double>>
-            weightedRatings =
-            ratings.join(weights) // [...]
+        weightedRatings =
+        ratings.join(weights); // [...]
```
Comment (zentol) [inline docs/content.zh/docs/dev/dataset/transformations.md:1201]:
```suggestion
        ratings.join(weights);
```

File: docs/content.zh/docs/dev/datastream/operators/joining.md [hunk 7]
```diff
@@ -277,11 +277,9 @@ orangeStream
     .intervalJoin(greenStream.keyBy(elem => /* select key */))
     .between(Time.milliseconds(-2), Time.milliseconds(1))
     .process(new ProcessJoinFunction[Integer, Integer, String] {
-        override def processElement(left: Integer, right: Integer, ctx: ProcessJoinFunction[Integer, Integer, String]#Context, out: Collector[String]): Unit = {
-         out.collect(left + "," + right); 
-        }
-      });
-    });
+        override def processElement(left: Integer, right: Integer, ctx: ProcessJoinFunction[Integer, Integer, String]#Context, out: Collector[String]): Unit = out.collect(left + "," + right)
```
Comment (zentol) [inline docs/content.zh/docs/dev/datastream/operators/joining.md:280]: move collect call to separate line

File: docs/content/docs/dev/datastream/user_defined_functions.md [hunk 8]
```diff
@@ -129,8 +133,8 @@ you can write
 
 ```scala
 class MyMapFunction extends RichMapFunction[String, Int] {
-  def map(in: String):Int = { in.toInt }
-};
+  def map(in: String): Int = in.toInt
```
Comment (zentol) [inline docs/content/docs/dev/datastream/user_defined_functions.md:136]: inconsistent use of spaces after `:`

File: docs/content.zh/docs/dev/datastream/user_defined_functions.md [hunk 9]
```diff
@@ -136,8 +140,8 @@ data.map { x => x.toInt }
 
 ```scala
 class MyMapFunction extends RichMapFunction[String, Int] {
-  def map(in: String):Int = { in.toInt }
-};
+  def map(in: String): Int = in.toInt
```
Comment (zentol) [inline docs/content.zh/docs/dev/datastream/user_defined_functions.md:143]: inconsistent use of spaces after `:`

File: docs/content.zh/docs/dev/datastream/user_defined_functions.md [hunk 10]
```diff
@@ -77,16 +77,20 @@ data.reduce((i1,i2) -> i1 + i2);
 
 ```java
 class MyMapFunction implements MapFunction<String, Integer> {
-  public Integer map(String value) { return Integer.parseInt(value); }
-};
+  public Integer map(String value) {
+    return Integer.parseInt(value);
+  }
```
Comment (zentol) [inline docs/content.zh/docs/dev/datastream/user_defined_functions.md:82]: let's move this back to a single line; also applies to other instances in this file, the english version, and I think 1 other page.

**Review by zentol — CHANGES_REQUESTED** [review]

File: docs/content/docs/dev/datastream/user_defined_functions.md (hunk 8 above)
Comment (paul8263) [reply docs/content/docs/dev/datastream/user_defined_functions.md:132]: Hi @zentol ,
Thank you for your reviewing this issue. I have made it consistent by adding one space after `:` .

**Pull request merged**